core/src/labels/label.h

```diff
@@ -186,6 +191,8 @@ class Label {
 
     int m_anchorIndex;
 
+    uint32_t m_selectionColor;
```

Comment:
I went on this path, but this need a pure virtual methods reimplemented on both, it's a little simpler as it is now.